std/process.d

```diff
@@ -4194,7 +4194,8 @@ else version (Posix)
             //args[0] = "x-www-browser".ptr;  // doesn't work on some systems
             args[0] = "xdg-open".ptr;
 
-        args[1] = url.tempCString();
+        const buffer = url.tempCString();
```

Comment:
Plea

Reply:
Sure.

Reply:
I think that the usage of `tempCString` is wrong here. The variable is a hack and shoudn't be necessary. It really isn't obvious why the extra variable is needed and the documentation of `tempCString` is lacking, to say the least. 

Why not use `std.string.toStringz` [1] instead?

[1] https://dlang.org/phobos/std_string.html#.toStringz

Reply:
It looks right to me. AIUI, `tempCString` returns an object which owns the string, but also is implicitly convertible to the string it owns. It is a dangerous pattern (which is why it's not public, I assume). So, if you want the string to live for longer than a D statement, then it has to be given a lifetime, such as saving it to a variable.

Reply:
`toStringz` always allocates a copy on the heap while `tempCString` prefers an internal buffer for small strings (< 256).
The C-string doesn't need to outlive this function, so `toStringz` is unnecessary here.

Reply:
Ok, I just thought that using `toStringz` achieves the same result but in a safer manner.